src/domain/settings/return-reasons/create-reason-modal.tsx

```diff
@@ -0,0 +1,84 @@
+import { useAdminCreateReturnReason } from "medusa-react"
+import React, { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import Button from "../../../components/fundamentals/button"
+import Input from "../../../components/molecules/input"
+import Modal from "../../../components/molecules/modal"
+import useToaster from "../../../hooks/use-toaster"
+
+// the reason props is used for prefilling the form when duplicating
+const CreateReturnReasonModal = ({ handleClose, reason }) => {
+  const { register, handleSubmit, reset } = useForm()
+  const toaster = useToaster()
+  const createRR = useAdminCreateReturnReason()
+
+  useEffect(() => {
+    reset({
+      value: reason.value,
```

Comment:
This causes errors when trying to create a return reason and causes the modal to crash, either this should be `reason?.value` or a value like `""` should be provided. 

I'm not quite sure that I understand why we pass a return reason to the create modal and why we reset every time we update the return reason

Reply:
good catch! As I mentioned in my PR, the `CreateReturnReasonModal` is used in two scenarios: 
1. Creating a return reason
2. Duplicating a return reason

For 2. a reason is passed to prefill the input fields, the user can then make the necessary changes and click "create" to create a new return reason. Hope this explains why the component takes a `reason` as prop. I added types for the props and renamed `reason` to `initialReason` to hopefully make this less confusing.

The useEffect would only run once when the `reason` prop is passed down to set the form values, but I think this doesn't communicate the intent in a good way, so I removed that and used `react-hook-form`'s `defaultValues`, which is the idiomatic way of prefilling form fields on initial render. Hope this clears the confusion a bit. 

Let me know what you think and thanks for the comments!

Reply:
ah ofc. that makes sense 😄  i must have missed it above, I think thats a very nice way of solving it and communicating the intent